flintrock/flintrock.py

```diff
@@ -297,6 +305,21 @@ def launch(
         raise UnsupportedProviderError(provider)
 
 
+def get_last_commit_sha(git_repository: str):
+    """
+    Retrieve the last commit's SHA from a repository hosted on Github.
+    """
+    parsed_url = urllib.parse.urlparse(git_repository)
+    if parsed_url.netloc != "github.com":
+        raise UsageError('Error: Retrieving the last commit\'s SHA only works with '
+                         'repositories hosted on Github.')
+    owner_repo = parsed_url.path[:-4] if parsed_url.path.endswith(".git") else parsed_url.path
```

Comment:
Yup, as part of this PR simply remove all mention of `.git` in the repo name if we don't need it, and remove the logic to strip it.